modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rule/LogicalScanConverterRule.java

```diff
@@ -81,6 +81,12 @@
                     .replace(collation);
 
                 Set<CorrelationId> corrIds = RexUtils.extractCorrelationIds(rel.condition());
+
+                if (!F.isEmpty(rel.projects())) {
```

Comment:
Why only for index scan? Table scan should be fixed too.

Reply:
Yep. IDE's Misscommit. Fixed.